Sources/NIOConcurrencyHelpers/lock.swift

```diff
@@ -190,16 +184,18 @@ public final class ConditionLock<T: Equatable> {
                 fatalError("caught error \(e) when calling pthread_cond_timedwait")
             }
         }
+        return true
     }
 
     /// Release the lock, setting the state variable to `newValue`.
     ///
     /// - Parameter newValue: The value to give to the state variable when we
     ///     release the lock.
     public func unlock(withValue newValue: T) {
+        self.lock()
```

Comment:
this doesn't seem right. The caller must have the lock locked before calling `unlock`. This should actually dead lock if used correctly.